packages/ifpack2/example/BlockTriDi.cpp

```diff
@@ -128,50 +145,112 @@ Teuchos::RCP<Tpetra::BlockCrsMatrix<Scalar, LocalOrdinal, GlobalOrdinal, Node> >
   using Teuchos::rcp;
   using Teuchos::rcp_dynamic_cast;
 
-  // Thanks for the code, Travis!
-  
   // Make the graph
   RCP<Xpetra::Matrix<Scalar, LocalOrdinal, GlobalOrdinal, Node> > FirstMatrix = BuildMatrix<Scalar, LocalOrdinal, GlobalOrdinal, Node>(matrixList,comm);
   RCP<const Xpetra::CrsGraph<LocalOrdinal, GlobalOrdinal, Node> > FGraph = FirstMatrix->getCrsGraph();
   
-  int blocksize = 3;
+  const int blocksize = matrixList.get("blockSize", 3);
   RCP<const Xpetra::TpetraCrsGraph<LocalOrdinal, GlobalOrdinal, Node> > TGraph = rcp_dynamic_cast<const Xpetra::TpetraCrsGraph<LocalOrdinal, GlobalOrdinal, Node> >(FGraph);
   RCP<const Tpetra::CrsGraph<LocalOrdinal,GlobalOrdinal,Node> > TTGraph = TGraph->getTpetra_CrsGraph();
   
-  RCP<Tpetra::BlockCrsMatrix<Scalar, LocalOrdinal, GlobalOrdinal, Node> > bcrsmatrix = rcp(new Tpetra::BlockCrsMatrix<Scalar, LocalOrdinal, GlobalOrdinal, Node> (*TTGraph, blocksize));
+  using BCRS = Tpetra::BlockCrsMatrix<Scalar, LocalOrdinal, GlobalOrdinal, Node>;
+  RCP<BCRS> bcrsmatrix = rcp(new BCRS (*TTGraph, blocksize));
   
-  const Tpetra::Map<LocalOrdinal, GlobalOrdinal, Node>& meshRowMap = *bcrsmatrix->getRowMap();
   const Scalar zero   = Teuchos::ScalarTraits<Scalar>::zero();
   const Scalar one   = Teuchos::ScalarTraits<Scalar>::one();
   const Scalar two   = one+one;
   const Scalar three = two+one;
   
   Teuchos::Array<Scalar> basematrix(blocksize*blocksize, zero);
-  basematrix[0] = two;
-  basematrix[2] = three;
-  basematrix[3] = three;
-  basematrix[4] = two;
-  basematrix[7] = three;
-  basematrix[8] = two;
   Teuchos::Array<Scalar> offmatrix(blocksize*blocksize, zero);
-  offmatrix[0]=offmatrix[4]=offmatrix[8]=-1;
-  
-  Teuchos::Array<LocalOrdinal> lclColInds(1);
-  for (LocalOrdinal lclRowInd = meshRowMap.getMinLocalIndex (); lclRowInd <= meshRowMap.getMaxLocalIndex(); ++lclRowInd) {
-    lclColInds[0] = lclRowInd;
-    bcrsmatrix->replaceLocalValues(lclRowInd, lclColInds.getRawPtr(), &basematrix[0], 1);
-    
-    // Off diagonals
-    if(lclRowInd > meshRowMap.getMinLocalIndex ()) {
-      lclColInds[0] = lclRowInd - 1;
-      bcrsmatrix->replaceLocalValues(lclRowInd, lclColInds.getRawPtr(), &offmatrix[0], 1);
+  int entryIdx = 0;
+  for(int rowIdx = 0; rowIdx < blocksize; ++rowIdx) {
+    for(int colIdx = 0; colIdx < blocksize; ++colIdx) {
+      entryIdx = rowIdx*blocksize + colIdx;
+      basematrix[entryIdx] = ((entryIdx % 2) == 0 ? one : two);
+    }
+    basematrix[rowIdx*blocksize + rowIdx] = three + 26 * one + blocksize * two;
```

Comment:
Since this an example, it would be good to have comment to explain this formula.

Reply:
Does a comment like this make sense to you?

```
    // We enforce that the block matrix is SPD by setting the diagonal value
    // to a value which is greater than the sum of the absolute value of all the
    // entries on the corresponding row knowing that the off diagonal blocks are
    // set to minus identity.
```

Reply:
@kliegeois Yes, this is also clear.